application/models/User.php

```diff
@@ -937,6 +937,19 @@ public function search()
         ));
     }
 
+    /** @inheritdoc */
+    public function scopes()
+    {
+        return array(
+            'active' => array(
+                'condition' => "expires > :now OR expires IS NULL",
```

Comment:
Yo @Shnoulle , this break upgrade from LS 3 to 5, since the `expires` column does not exists when the upgrade starts.

Reply:
The upgrade procedure must not use any ActiveRecord models. They cache their columns etc, it's not reliable.

Reply:
OK, then broke before upgrade happen.

DB version dependant scope ? OK for this fix ?

Reply:
@ptelu can discuss

Reply:
I don't see how do it except a version dependant scope … i create a PR for master … OK ?

Reply:
> Maybe you can add the scope dependant on the version?

?

Reply:
@olleharstedt  https://github.com/LimeSurvey/LimeSurvey/pull/3607

I test an upgrade connected and not connected from 3.X : seems OK.

Reply:
> Maybe you can add the scope dependant on the version?

Oh, got it :) Yes an alternative.  I think scope dependant allow to use it on other code in future :)